grammar/test/FloatingPointExpression.g4

```diff
@@ -0,0 +1,88 @@
+grammar FloatingPointExpression;
+
+expression
+
+  : arithmeticExpression EOF;
+
+primaryExpression
+
+  : constant
+
+  | '(' arithmeticExpression ')';
+
+unaryOperator
+
+  : '+' | '-';
+
+unaryExpression
+
+  : primaryExpression
+
+  | unaryOperator unaryExpression;
+
+multiplicativeExpression
+
+  : unaryExpression
+
+  | multiplicativeExpression '*' unaryExpression
+
+  | multiplicativeExpression '/' unaryExpression;
+
+additiveExpression
+
+  : multiplicativeExpression
+
+  | additiveExpression '+' multiplicativeExpression
+
+  | additiveExpression '-' multiplicativeExpression;
+
+arithmeticExpression
+
+  : additiveExpression;
+
+argumentExpressionList
+
+  : arithmeticExpression (',' arithmeticExpression)*;
+
+constant
+
+  : decimalFloatingConstant
+
+  | symbolicConstant;
+
+decimalFloatingConstant
+
+  : DecimalFloatingConstant;
+
+symbolicConstant
+
+  : 'pi';
+
+DecimalFloatingConstant
+
+  : FractionalConstant ExponentPart?
+
+  | DigitSequence ExponentPart?;
+
+DigitSequence
+
+  : [0-9]+;
+
+WhiteSpace
+
+  : [ \r\n\t] + -> skip;
+
+fragment FractionalConstant
+
+  : DigitSequence? '.' DigitSequence
+
+  | DigitSequence '.';
+
+fragment ExponentPart
+
+  : 'e' ('+' | '-')? DigitSequence
+
+  | 'E' ('+' | '-')? DigitSequence;
+
+// Handle characters which failed to match any other token
```

Comment:
Explain again why we should do this?

Reply:
If we do not add this here, then the lexer throws a (potentially silent) error, but still produces the remaining tokens and sends them to the parser.  So an expression like "2PI" would be given to the parser as "2", and the parser would succeed.  If we added the remaining tokens, the parser would be given "2", "P", "I", and spit out an exception when it got to "P" because that doesn't work in the expression.

This isn't strictly necessary, as we could add a custom error handler to the lexer, but we'd also have to remove some of the default error handlers the lexer has, which would suppress other potentially useful errors.   I had seen this recommended on StackOverflow (https://stackoverflow.com/questions/22415208/get-rid-of-token-recognition-error), see the first solution recommended.